DTM serial port connection documentation (#4040)
nrfconnect/sdk-nrf: samples/bluetooth/direct_test_mode/README.rst

   However, you must still program the application core to boot up the network core.
   You can use any sample for this, for example, the :ref:`nrf5340_empty_app_core`.

.. _dtm_testing:

Direct Test Mode terminal connection
************************************

> Follow the instructions in this section to connect with the DK which is running the DTM to send commands and receive responses.

Reviewer: ```suggestion
To run Direct Test Mode (DTM) to send commands and receive responses, you can connect to your development kit with RealTerm in Windows or Minicom in Linux.
```

Author: Done

The Bluetooth Low Energy DTM UART interface standard specifies the following configuration:

* Eight data bits
* No parity
* One stop bit
* No hardware flow control
* A selection of bit rates from 9600 to 1000000, one of which must be supported by the DUT.
  It might be possible to run other bit rates by experimenting with parameters.

.. note::
   The default bit rate of the DTM UART driver is 19200 bps, which is supported by most certified testers.

You have to send all commands as a two-byte Hex number. The responses must have the same format.

Connect with RealTerm (Windows)
===============================

> RealTerm terminal program offers a graphical interface for setting up your connection.

Reviewer: ```suggestion
The RealTerm terminal program offers a graphical interface for setting up your connection.
```

Author: Done

.. figure:: /images/realterm.png
   :alt: RealTerm start window

>    The RealTerm start window.

Reviewer: No full stop in the captions.

Author: Ok

   .. figure:: /images/real_term_serial_port.png
      :alt: RealTerm serial port settings

3. Open the :guilabel:`Send` tab:

   a. Write the command as Hex number in the field **(1)**.
      For example, write **0x00 0x00** to send a **Reset** command.
   b. Click the :guilabel:`Send Numbers` button **(2)** to send the command.
   c. Observe the response in the DTM in area **(3)**.
      The response is encoded as Hex numbers.

   .. figure:: /images/realterm_commands.png
      :alt: RealTerm commands sending

Connect with Minicom (Linux)
============================

Minicom is a serial communication program that connects to the DTM device

On the Linux operating system, install a Minicom terminal.
On Ubuntu, run:

.. code-block::

   sudo apt-get install minicom

1. Run the Minicom terminal:

   .. code-block::

      sudo minicom -D <DTM serial port> -s

   For example:

   .. code-block::

      sudo minicom -D /dev/serial/by-id/usb-SEGGER_J-Link_000683580193-if00 -s

   The **-s** option switches you to Minicom setup mode.

2. Configure the Minicom terminal:

   .. figure:: /images/minicom_setup_window.png
      :alt: minicom configuration window

>       Configuration window.

Reviewer: Leave out the full stop.

Author: Ok

1. Connect both development kits to the computer using a USB cable.
   The computer assigns to the development kit a COM port on Windows or a ttyACM device on Linux, which is visible in the Device Manager.
#. Connect to both kits with a terminal emulator.

>    See :ref:`Direct Test Mode terminal connection` for the required settings.

Reviewer: ```suggestion
   See :ref:`Direct Test Mode terminal connection`_ for the required settings.
```

Author: Fixed

1. Connect the development kit to the computer using a USB cable.
   The computer assigns to the development kit a COM port on Windows or a ttyACM device on Linux, which is visible in the Device Manager.
#. Connect to the kit with a terminal emulator.

>    See :ref:`Direct Test Mode terminal connection` for the required settings.

Reviewer: ```suggestion
   See :ref:`Direct Test Mode terminal connection`_ for the required settings.
```

Author: Changed. It should be `Direct Test Mode terminal connection`_ without ":ref:"

Reviewer: You are absolutely right.

#. Swap roles.
   Set the application to the RX mode and the connected development kit to the TX mode.

Direct Test Mode terminal connection
------------------------------------

> To run Direct Test Mode (DTM) to send commands and receive responses, you can connect to your development kit with RealTerm in Windows or Minicom in Linux.

Reviewer 2: Can you run DTM? This sounds strange to me.
Maybe "To use the Direct Test Mode sample to send ..." or "To send commands to and receive responses from the development kit that runs the Direct Test Mode sample, connect to it with ..."

Author: Yes you right. This sounds a little strange. I used your second suggestion.

The Bluetooth Low Energy DTM UART interface standard specifies the following configuration:

* Eight data bits
* No parity
* One stop bit
* No hardware flow control
* A selection of bit rates from 9600 to 1000000, one of which must be supported by the DUT.
  It might be possible to run other bit rates by experimenting with parameters.

.. note::
   The default bit rate of the DTM UART driver is 19200 bps, which is supported by most certified testers.

> You have to send all commands as two-byte Hex numbers. The responses must have the same format.

Reviewer 2: ```suggestion
You must send all commands as two-byte HEX numbers. 
The responses must have the same format.
```

Author: Done

> You have to send all commands as a two-byte Hex number. The responses must have the same format.

Reviewer: ```suggestion
You have to send all commands as two-byte Hex numbers. The responses must have the same format.
```

Author: Done

Connect with RealTerm (Windows)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

The RealTerm terminal program offers a graphical interface for setting up your connection.

.. figure:: /images/realterm.png
   :alt: RealTerm start window

   The RealTerm start window

> To test the DTM with RealTerm, complete the following steps:

Reviewer: ```suggestion
To test DTM with RealTerm, complete the following steps:
```

Author: Done

> 1. On the :guilabel:`Display` tab, set :guilabel:`Display As` to :guilabel:`Hex[space]`.

Reviewer 2: Empty line before the first step.

Author: Ok. Added

   .. figure:: /images/realterm_hex_display.png
      :alt: Set the RealTerm display format

> 2. Open the :guilabel:`Port` tab and configure the serial port parameters:

Reviewer 2: #. instead of 2., 3. in all steps except the first.

Author: Done

>    a. Set the :guilabel:`Baud` to 19200 **(1)**

Reviewer 2: Periods missing in this list.

Author: Added

>    b. Select your Jlink serial port from the :guilabel:`Port` list **(2)**

Reviewer 2: Same here - only a., and then #. for the remaining steps.

Author: Done

>    b. Select your Jlink serial port from the :guilabel:`Port` list **(2)**

Reviewer 2: J-Link

Author: Done

   c. Set the port status to "Open" **(3)**

   .. figure:: /images/real_term_serial_port.png
      :alt: RealTerm serial port settings

3. Open the :guilabel:`Send` tab:

>    a. Write the command as Hex number in the field **(1)**.

Reviewer: ```suggestion
   a. Write the command as a Hex number in the field **(1)**.
```

Author: Done

      For example, write **0x00 0x00** to send a **Reset** command.
   b. Click the :guilabel:`Send Numbers` button **(2)** to send the command.
   c. Observe the response in the DTM in area **(3)**.

>       The response is encoded as Hex numbers.

Reviewer 2: ```suggestion
      The response is encoded as HEX numbers.
```
or 
```suggestion
      The response is encoded as hexadecimal numbers.
```

Reviewer 2: Same in line 320 and 379.

Author: Done

   .. figure:: /images/realterm_commands.png
      :alt: RealTerm commands sending

Connect with Minicom (Linux)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> Minicom is a serial communication program that connects to the DTM device

Reviewer: ```suggestion
Minicom is a serial communication program that connects to the DTM device.
```

Author: Done

On the Linux operating system, install a Minicom terminal.
On Ubuntu, run:

.. code-block::

   sudo apt-get install minicom

1. Run the Minicom terminal:

   .. code-block::

>       sudo minicom -D <DTM serial port> -s

Reviewer 2: We shouldn't use <> for variables.
Use italics instead, and .. parsed-literal:: to make the italics display in a code block.
See for example the "Configuring wpantund" section here: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/_sources/ug_thread_tools.rst.txt

Author: Sure. Done

   For example:

   .. code-block::

      sudo minicom -D /dev/serial/by-id/usb-SEGGER_J-Link_000683580193-if00 -s

   The **-s** option switches you to Minicom setup mode.

2. Configure the Minicom terminal:

   .. figure:: /images/minicom_setup_window.png
      :alt: minicom configuration window

      Configuration window

   a. Select :guilabel:`Serial port setup` and set UART baudrate to 19200.

      .. figure:: /images/minicom_serial_port.png
         :alt: minicom serial port settings

   b. Select :guilabel:`Screen and keyboard` and press **S** on the keyboard to enable the **Hex Display**.
   c. Press **Q** on the keyboard to enable **Local echo**.

      .. figure:: /images/minicom_terminal_cfg.png
         :alt: minicom terminal screen and keyboard settings

   Minicom is now configured for receiving data.
   However, you cannot use it for sending DTM commands.

3. Send DTM commands:

   To send DTM commands, use **echo** with **-ne** options in another terminal.

>    You must encode the data as Hex numbers (\xHH, byte with hexadecimal value HH (1 to 2 digits)).

Reviewer 2: no double parentheses ;)

Author: Ok

#. Swap roles.
   Set the application to the RX mode and the connected development kit to the TX mode.

Direct Test Mode terminal connection
------------------------------------

To send commands to and receive responses from the development kit that runs the Direct Test Mode sample, connect to it with RealTerm in Windows or Minicom in Linux.

The Bluetooth Low Energy DTM UART interface standard specifies the following configuration:

* Eight data bits
* No parity
* One stop bit
* No hardware flow control
* A selection of bit rates from 9600 to 1000000, one of which must be supported by the DUT.
  It might be possible to run other bit rates by experimenting with parameters.

.. note::
   The default bit rate of the DTM UART driver is 19200 bps, which is supported by most certified testers.

You must send all commands as two-byte HEX numbers. 
The responses must have the same format.

Connect with RealTerm (Windows)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

The RealTerm terminal program offers a graphical interface for setting up your connection.

> .. figure:: /images/realterm.png

Reviewer 2: The first two pictures (echo.png and minicom.png) don't seem to be used anywhere - should remove.

Author: Ok removed

On Ubuntu, run:

.. code-block:: console

   sudo apt-get install minicom

1. Run the Minicom terminal:

   .. parsed-literal::
      :class: highlight

      sudo minicom -D *DTM serial port* -s

   For example:

   .. code-block:: console

      sudo minicom -D /dev/serial/by-id/usb-SEGGER_J-Link_000683580193-if00 -s

   The **-s** option switches you to Minicom setup mode.

#. Configure the Minicom terminal:

   .. figure:: /images/minicom_setup_window.png
      :alt: minicom configuration window

      Configuration window

   a. Select :guilabel:`Serial port setup` and set UART baudrate to 19200.

      .. figure:: /images/minicom_serial_port.png
         :alt: minicom serial port settings

   #. Select :guilabel:`Screen and keyboard` and press **S** on the keyboard to enable the **Hex Display**.
   #. Press **Q** on the keyboard to enable **Local echo**.

      .. figure:: /images/minicom_terminal_cfg.png
         :alt: minicom terminal screen and keyboard settings

   Minicom is now configured for receiving data.
   However, you cannot use it for sending DTM commands.

#. Send DTM commands:

   To send DTM commands, use **echo** with **-ne** options in another terminal.

>    You must encode the data as Hex numbers (\xHH, byte with hexadecimal value HH, 1 to 2 digits).

Reviewer 2: HEX or hexadecimal

Author: Done
